Checkout/Checkout/Source/Validation/Validators/CardValidator.swift

```diff
@@ -95,10 +104,20 @@ public class CardValidator: CardValidating {
     expiryYear: String
   ) -> Result<ExpiryDate, ValidationError.ExpiryDate> {
     logManager.queue(event: .validateExpiryString)
-    return validateExpiry(
-      expiryMonth: expiryMonth,
-      expiryYear: expiryYear
-    )
+
+    guard let expiryMonth = Int(expiryMonth) else {
+      return .failure(.invalidMonthString)
+    }
+
+    switch fourDigitYear(from: expiryYear) {
+    case .success(let expiryYear):
+      return validateExpiry(
+        expiryMonth: expiryMonth,
+        expiryYear: expiryYear
+      )
+    case .failure(let error):
+      return .failure(error)
+    }
```

Comment:
Is the same code logic like the block code from line 131 to 146?

Reply:
yeah looks duplicated to some degree. have cleaned up